examples/hello_world/CMakeLists.txt

```diff
@@ -2,6 +2,17 @@ cmake_minimum_required(VERSION 3.12)
 
 set(ROOT_DIR ${CMAKE_CURRENT_LIST_DIR}/../../)
 set(CMAKE_TOOLCHAIN_FILE ${ROOT_DIR}/toolchain-rv32.cmake)
+# ESP Target MCU. Supported: esp32c3, esp32h2
+list(APPEND supported_mcu esp32c3 esp32h2)
+
+if(target IN_LIST supported_mcu)
+    message(STATUS "Configure project for ${target} ... ")
+else ()
+    message(FATAL_ERROR "Supported target ${supported_mcu}")
+endif()
+string(TOUPPER ${target} target_uppercase)
+set(${target_uppercase} 1)
+configure_file(${ROOT_DIR}/config.h.in ${ROOT_DIR}/config.h)
```

Comment:
If the place where the defines from config.h are used is the `common` library, then it looks a bit odd that the responsibility for generating the config file falls onto the example CMakeLists file. Should this happen in common CMakeLists file, instead, if that's the place where the config file is used?

Reply:
I am not sure. 

I made a dependency from target also in utils.cmake (during the ld-file selection), common and hal. 
So, all of the examples configure target and this target then used in other parts. 

I assume, that copy-paste this lines is from one example list file to another is not an option, so probably I can move that target-selection logic to configure.cmake file and include it in every file. 
What do you think?

Reply:
I think you can create a new cmake library, `config`, and make other libraries which use config.h dependent on `config`. The example can use `add_subdirectory(${ROOT_DIR}/config config)` to add `config` to the build and `hal` & `common` can use `target_link_libraries(... PRIVATE config)` to declare the dependency.

(This also makes sure we do not rely on a global variable getting set in one CMake library and then used in the other.)

Reply:
Hm....
Is it really necessary to create library only from one header file? 

Isn't that an over complication, when it is possible to use only `target_include_directories(... PRIVATE ${ROOT_DIR}/config)` to include `config.h` file to the all libraries (example, common, hal)?

Reply:
I think that would be worse for encapsulation: hal and common would have to "know" the location of the generated config directory.

Other than that, looks like currently config.h isn't used for anything, so you could sidestep the whole question entirely, for the time being?